community/kernel/src/main/java/org/neo4j/kernel/impl/constraints/ConstraintSemantics.java

```diff
@@ -36,7 +36,7 @@
  */
 public interface ConstraintSemantics
 {
-    void validateNodePropertyExistenceConstraint( Cursor<NodeItem> allNodes, int label, int propertyKey )
+    void validateNodePropertyExistenceConstraint( Iterator<Cursor<NodeItem>> allNodes, int label, int propertyKey )
```

Comment:
Yeah it is a bit silly.. but you need to "close" the "SingleNodeCursor" since they are cached on the statement. I agree this is not nice but we can clean this us later, it is not a small change.